source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp

```diff
@@ -72,6 +74,23 @@ namespace
         return {};
     }
 
+    void createEmptyOutputFile()
```

Comment:
This should return the file handle, opening an closing the file many times is inefficient and not very elegant.

Reply:
even though it means passing the file handle into quite nice functions, that doesn't seem to need them? In addition to the verbosity enum? I could make those changes and push as fixup. 

However, I would rather refactor the entire program, and have those nice benchmark functions return the raw results. We can for example add one layer like this (one example):
Existing:
- `benchmarkConnect(camera, numConnects);`

Proposal:
- 'benchmarkConnect(camera, numConnects, fileHandle, verbosityEnum);'
  - `printHeaderBenchmarkConnect(fileHandle, verbosityEnum)`
  - `results = benchmarkConnect(camera, numConnects)`
  - `processBenchmarkConnectResults(results fileHandle, verbosityEnum)`

I would then pull out any processing of the results from `benchmarkConnect(camera, numConnects)`

Reply:
If you do the verbose cli options my proposal is to not deal with files in here at all, just dump all to std out. Then this function goes away.

To answer your question, if you were to keep this function, I would have created a Printer class that opens the file in ctor and closes it in dtor and then all the nice functions you talk about would be member functions on that class.

Reply:
To avoid passing the verbosity level around you could still make the Printer class, but not pass it a file name, but instead pass it the verbosity level in ctor. Then all functions can be moved into that class and signatures can be kept unchanged.

A printer class also makes it easier to add more customizations points later on.

Reply:
> 
> 
> If you do the verbose cli options, then my proposal is to not deal with files in this program at all, just dump everything to stdout. This function then goes away.

The idea was to allow customer to see the nice output, while we could still ask for the raw data. With your option he would have to run it twice, or be bothered with the verbose prints.

Reply:
@SatjaSivcev, what do you think about just writing to std::cout. I'm thinking we would only print what's currently going to the file when the verbosity is enabled. In other words it would be an OR operation. The instruction for the customer could then be to run 
```bash
./ZividBenchmark.exe --vv > zivid_benchmark_raw.txt
```

Reply:
One minus btw. `-vv` is shorthand for `-v -v` which is shorthand for `--verbose --verbose`. Just convention I have seen some places. Feel free to call it other things.

Reply:
The program would output raw data you add `-vv`? And if you don't, it would output just the mean and median as of now?

Reply:
Running twice or verbose prints is something I would avoid. If possible, I would avoid options, it is best if the customer has only one way to run the program and that gives them a nice print and us raw data. Can this be achieved with the above suggestion?

Reply:
I believe a compromise would be that we tag the data and use `| grep` or `| Select-string` (for Powershell). What do you think about that? This could be combined with a verbose option, so that the default execution only gives a nice print.

Reply:
Only with option @SatjaSivcev, but then yes. Tools dumping debug files to $PWD without me asking for it would annoy me a lot at least.

Keep it simple, no grepping, no magic. Verbose mode (`--verbose`) is very common or option to save file (`--save-debug-info file-name`) would be the two options that I feel is natural.

Think I sated my opinion. Others view on this? @knatten, @apartridge, @knzivid

Reply:
I agree with not touching the home directory. Running the benchmark twice would have counter-intuitive behavior (be it overwriting or appending to the existing file)

Reply:
@knzivid, thanks for your feedback! You seem to argue against running the benchmark twice, like we do. Yet I'm not sure if you argue against a `--verbose` option. You also seem to argue against `--save_to_file`. May I ask you to refine your opinion?

A `--verbose` option doesn't give a nice output, hence why we say it would lead to multiple runs.

A `--save-to-file` option seems like the best alternative to me, as it 
- allows for nice output in console
- gives us the raw data
- makes the customer aware that we're writing to $PWD

I did not receive any feedback on the data-tagging option. It's quite common to have [INFO], [WARNING] and [ERROR] in logs, right? I was thinking about only tagging the data-output, thus making it possible to run `--verbose` without actually seeing it. I mentioned grep, but it seems awk can do the job better in splitting the output. I'm not a fan of this option btw, I'd prefer `--save-to-file`

Reply:
My response to "data-tagging" was "Keep it simple, no grepping, no magic.".

Reply:
Ok, so we'll go for the `--save-to-file` option.

Reply:
Would call it, `--save-debug-info` or similar since it does not save the regular output.